src/formatDate.js

```diff
@@ -8,7 +8,37 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [yearKey, monthKey, dayKey, separator] = fromFormat;
+  const splittedDate = date.split(separator);
+
+  const oldDateObj = [yearKey, monthKey, dayKey].reduce((acc, key, index) => {
+    acc[key] = splittedDate[index];
+
+    return acc;
+  }, {});
+
+  const updatedDateObj = changeYearsFormat(oldDateObj, toFormat);
+
+  return toFormat
+    .slice(0, -1)
+    .map((format) => updatedDateObj[format])
+    .join(toFormat[3]);
+}
+
+function changeYearsFormat(dateObj, newFormat) {
+  const result = { ...dateObj };
+
+  if (result.YY && newFormat.includes('YYYY')) {
```

Comment:
You are checking if `result.YY` exists and if `newFormat` includes 'YYYY', which is good. However, you should also ensure that `result.YY` is a string representing a two-digit year to avoid potential runtime errors.